container/pymzqc-usecase/pymzqc-usecase.py

```diff
@@ -21,327 +21,586 @@
+                               pairing_file_name=tide_td_pair_file,
+                               decoy_prefix='DECOY_')
+        results = psms.assign_confidence(
+            score_column="xcorr score", desc=True, pep_fdr_type="peptide-only", threshold=fdr/100)
+        prt_df = results.confidence_estimates['proteins']
+        # add delta ppm per peptide and fix column names
+        pep_df = results.confidence_estimates['peptides'].rename(columns={"scan": "scan_id"}).merge(filtered_tide_target.rename(columns={"scan": "scan_id"})[
+            ['scan_id', 'sequence', 'charge', 'peptide mass', 'spectrum precursor m/z']], how="inner", on=['scan_id', 'sequence']).rename(columns={"spectrum precursor m/z": "experimentalMassToCharge"})
+        pep_df['calculatedMassToCharge'] = (
+            pep_df['peptide mass']+PROTON_AMU*pep_df['charge'])/pep_df['charge']
+        psm_df = results.confidence_estimates['psms'].rename(columns={"scan": "scan_id"})
+
+    run.tide_target_file = tide_target_file
+    run.tide_decoy_file = tide_decoy_file
+    run.tide_td_pair_file = tide_td_pair_file
+    run.crema_fdr = fdr
+    run.id_df = pep_df[pep_df['accept'] == True]
+    run.n_prot = prt_df[prt_df['accept'] == True]["protein id"].nunique()
+    run.n_pep = run.id_df["sequence"].nunique()
+    run.n_ms2id = psm_df[psm_df['accept'] == True]['scan_id'].nunique()
+
+    logging.debug("Registered proteins "+str(run.n_prot))
+    logging.debug("Registered peptides "+str(run.n_pep))
+    logging.debug("Registered id MS2 "+str(run.n_ms2id))
+    return run
+
+
+def construct_mzqc(run: Run, quality_metric_values: List[qc.QualityMetric]) -> qc.MzQcFile:
+    """
+    construct_mzqc will create a mzqc object to export 
+
+    Parameters
+    ----------
+    run : Run
+            The Run dataclass object with the necessary metadata
+
+    quality_metric_values : List[qc.QualityMetric]
+            The calculated metrics to be added to this run's mzqc representation
+
+    Returns
+    -------
+    qc.MzQcFile
+            An mzqc file object
+    """
+    infi1 = qc.InputFile(name=run.mzml_path, location=run.mzml_path,
+                         fileFormat=qc.CvParameter("MS:1000584", "mzML format"))
+    infi1.fileProperties.append(qc.CvParameter(
+        "MS:1003151", "SHA-256", run.checksum))
+    infi1.fileProperties.append(qc.CvParameter(
+        run.instrument_type.id, run.instrument_type.name))
+    infi1.fileProperties.append(qc.CvParameter(
+        "MS:1000747", "completion time", run.completion_time))
+    infi2 = qc.InputFile(name=run.tide_target_file, location=run.tide_target_file,
+                         fileFormat=qc.CvParameter("MS:1000914", "tab delimited text format"))
+    anso1 = qc.AnalysisSoftware(
+        accession="MS:1002575", name="Tide", version="4.2", uri="https://crux.ms/")
+    anso2 = qc.AnalysisSoftware(accession="MS:1003357", name="simple qc metric calculator",
```

Comment:
this ID is `ANN-SoLo`.
maybe go for
```
[Term]
id: MS:1003399
name: quality control software
def: "Software that creates or manipulates QC-related data." [PSI:MS]
is_a: MS:1001457 ! data processing software
```

Reply:
[Term]
id: MS:1003145
name: ThermoRawFileParser
def: "Cross-platform software to convert Thermo RAW files to a number of open formats." [DOI:10.1021/acs.jproteome.9b00328, PMID:31755270]
is_a: MS:1001457 ! data processing software

Reply:
totally missed that, review ftw